src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -1879,16 +2020,17 @@ impl Step for Assemble {
         //
         // FIXME: It may be faster if we build just a stage 1 compiler and then
         //        use that to bootstrap this compiler forward.
+        debug!(
+            "recursively ensuring build compiler is available: compiler(stage = {}, host = {:?})",
+            target_compiler.stage - 1,
+            builder.config.build,
+        );
```

Comment:
This will be printed multiple times. The log here is part of the recursion already.

```suggestion
        debug!(
            "ensuring build compiler is available: compiler(stage = {}, host = {:?})",
            target_compiler.stage - 1,
            builder.config.build,
        );
```

Reply:
Good point

Reply:
Removed the "recursively"